src/Tasks/Microsoft.NET.Build.Tasks/ResolvePackageDependencies.cs

```diff
@@ -150,11 +150,11 @@ internal ResolvePackageDependencies(LockFile lockFile, IPackageResolver packageR
         /// </summary>
         protected override void ExecuteCore()
         {
-            var targetFrameworkToAliasMap = LockFile.PackageSpec.TargetFrameworks.ToDictionary(tf => tf.FrameworkName.DotNetFrameworkName, tf => tf.TargetAlias);
+            var targetFrameworkToAliasMap = LockFile.PackageSpec.TargetFrameworks.ToDictionary(tf => ( tf.FrameworkName.DotNetFrameworkName, tf.FrameworkName.DotNetPlatformName ), tf => tf.TargetAlias);
 
             _targetNameToAliasMap = LockFile.Targets.ToDictionary(t => t.Name, t =>
             {
-                var alias = targetFrameworkToAliasMap[t.TargetFramework.DotNetFrameworkName];
+                var alias = targetFrameworkToAliasMap[(t.TargetFramework.DotNetFrameworkName, t.TargetFramework.DotNetPlatformName)];
```

Comment:
Can we get rid of `targetFrameworkToAliasMap` and just use `LockFileExtensions.GetLockFileTargetAlias`?

```suggestion
                var alias = LockFile.GetLockFileTargetAlias(t);
```

That is more likely to work correctly when we support generalized TargetFramework aliasing.